pkg/cluster/install.go

```diff
@@ -201,6 +201,8 @@ func (m *manager) Update(ctx context.Context) error {
 		steps.Action(m.renewMDSDCertificate),
 		steps.Action(m.updateOpenShiftSecret),
 		steps.Action(m.updateAROSecret),
+		steps.Action(m.restartAROOperatorMaster), // depends on m.updateOpenShiftSecret; the point of restarting is to pick up any changes made to the secret
```

Comment:
do we need to wait for the reconciliation of the credentialsRequest before we restart it?

Reply:
That's a good point. I think we want to, but I'll have to think about how to check that the reconciliation is done.

Reply:
What would happen if the credentialsRequest hadn't reconciled? I think that would cause the ARO operator to crashloop briefly until it's finished.

Reply:
I did somewhat of a deep dive while figuring out how to go about this. The CredentialsRequest's [status](https://docs.openshift.com/container-platform/4.12/rest_api/security_apis/credentialsrequest-cloudcredential-openshift-io-v1.html#status) has a `lastSyncCloudCredsSecretResourceVersion` that we could check, but it feels hacky because we would have to save the old version prior to updating the secret so that we can see what it used to be. I'm thinking we can just use the `lastSyncTimestamp` instead by checking to see if it's something within the past 5 minutes. Your thoughts?

It's also tricky because CredentialsRequests haven't made their way into openshift/client-go. Based on my reading, I'll have to use https://pkg.go.dev/k8s.io/client-go/dynamic#DynamicClient to work with the CredentialsRequest from within `pkg/cluster`, but let me know if there's a better way.

Reply:
I pushed a solution that uses the `lastSyncTimestamp`, and I feel like this should be fine.